homeassistant/components/bom/weather.py

```diff
@@ -54,7 +54,9 @@ def name(self):
     @property
     def condition(self):
         """Return the current condition."""
-        return self.bom_data.get_reading("weather")
+        return self.bom_data.get_reading("weather") or self.bom_data.get_reading(
```

Comment:
does this need to check `if self.bom_data.get_reading("weather") == "-"`?

Reply:
No, as `None` is returned in that context, thus it returns the latter reading of the `OR` statement. 

See here: https://github.com/home-assistant/core/blob/5dcaeebdac29c076a0ebf0ba779a3bcf9221b516/homeassistant/components/bom/sensor.py#L223